pkg/ovs/openflow/interfaces.go

```diff
@@ -208,6 +209,7 @@ type Action interface {
 	LoadPktMarkRange(value uint32, to *Range) FlowBuilder
 	LoadIPDSCP(value uint8) FlowBuilder
 	LoadRange(name string, addr uint64, to *Range) FlowBuilder
+	LoadInPort(value uint16) FlowBuilder
```

Comment:
@luolanzone Why we need to override in_port? The field should be filled by OVS itself.
@wenyingd is it legal/common to rewrite in_port?

Reply:
We use LoadInPort to write in_port as 0, so the traffic can be resent to the coming port. In the Gateway Node, the reply traffic may go to the Pod in current Gateway or go to the regular Node, so we need to clean up this in_port, otherwise,it will be dropped when it's for regular Node because both in and out traffic are from the same tunnel port.
An example is as below which is from [ovs-fields.txt](http://www.openvswitch.org/support/dist-docs/ovs-fields.7.txt)
`For  example,  actions=load:0->NXM_OF_IN_PORT[],out‐
       put:123  will  output  to  port  123 regardless of whether it is in the
       ingress port.`

Reply:
We resolved this issue by adding a flow to `output: inPort`, does it work for this case?

Reply:
It doesn't work in this case because the packet's out put might be in_port, or different port if its destination is a Pod in current Gateway.

Reply:
The flow that uses `output: inport` should check the packet comes from and goes to same tunnel port, it shouldn't apply for all flows, for example:
```
priority=210, in_port=5, reg2 (storing output_port)=5 actions:output:inPort
priority=200, actions:output:reg2
```
The point is, the flow should check this hairpin case is intended, instead of erasing the in_port unconditionally to workaround an intended restriction added by OVS.

Reply:
It actually hit the flow `table=Output,priority=200,reg0=0x100/0x100 actions=output:NXM_NX_REG1[]`, When the packet is from tunnel port and goes to a regular Node,  the output is 1 after it goes through L3Forwarding and L2ForwardingCalc table, because in_port is erased, so it can forward correctly, otherwise, it will be dropped because it's the same as in_port.

Reply:
Add comments to explain when this is used.

Reply:
> It actually hit the flow `table=Output,priority=200,reg0=0x100/0x100 actions=output:NXM_NX_REG1[]`, When the packet is from tunnel port and goes to a regular Node, the output is 1 after it goes through L3Forwarding and L2ForwardingCalc table, because in_port is erased, so it can forward correctly, otherwise, it will be dropped because it's the same as in_port.

I don't see why it's necessary to erase in_port from this explanation. I understand the packet will be dropped if output port is same as in port, but I have explained how it was resolved in other antrea's case and why it's not preferred to erase in_port. Could you check my comments above to see if it can resolve your case? In my understanding, you could simply add one flow:
```
table=Output,priority=210,reg0=0x100/0x100, in_port=tunnel_port, reg1=tunnel_port actions=output:inPort
```

Reply:
Added a new flows with suggested one, I didn't think of an efficient way before to use reg0 to distinguish local Pod and remote one. I have removed the LoadInPort function, thanks. @tnqn